src/GraphQL.Tests/Instrumentation/OpenTelemetryTests.cs

```diff
@@ -0,0 +1,384 @@
+
+    [Fact]
+    public async Task Extensibility()
+    {
+        var executionOptions = new ExecutionOptions
+        {
+            Query = "query helloQuery { hello }",
+            RequestServices = _host.Services,
+        };
+
+        _options.EnrichWithExecutionOptions = (activity, options) =>
+        {
+            options.ShouldBe(executionOptions);
+            activity.SetTag("testoptions", "test1");
+        };
+        _options.EnrichWithDocument = (activity, options, schema, document, operation) =>
+        {
+            options.ShouldBe(executionOptions);
+            schema.AllTypes["Query"].ShouldBeOfType<AutoRegisteringObjectGraphType<Query>>();
+            document.Source.ToString().ShouldBe(executionOptions.Query);
+            operation.Operation.ShouldBe(GraphQLParser.AST.OperationType.Query);
+            activity.SetTag("testdocument", "test2");
+        };
+        _options.EnrichWithExecutionResult = (activity, options, result) =>
+        {
+            options.ShouldBe(executionOptions);
+            result.Query.ToString().ShouldBe(executionOptions.Query);
+            activity.SetTag("testresult", "test3");
+        };
+
+        // execute GraphQL document
+        var result = await _executer.ExecuteAsync(executionOptions).ConfigureAwait(false);
+
+        // verify GraphQL response
+        _serializer.Serialize(result).ShouldBe("""{"data":{"hello":"World"}}""");
+
+        // verify activity telemetry
+        var activity = _exportedActivities.ShouldHaveSingleItem();
+        activity.Tags.ShouldBe(new KeyValuePair<string, string>[]
+        {
+            new("graphql.document", "query helloQuery { hello }"),
+            new("testoptions", "test1"),
+            new("graphql.operation.type", "query"),
+            new("graphql.operation.name", "helloQuery"), // operation name pulled from document
+            new("testdocument", "test2"),
+            new("testresult", "test3"),
+        });
+        activity.DisplayName.ShouldBe("query helloQuery");
+#if NET6_0_OR_GREATER
+        activity.Status.ShouldBe(ActivityStatusCode.Unset);
+#endif
+    }
+
+    [Fact]
+    public async Task Filterable()
+    {
+        var executionOptions = new ExecutionOptions
+        {
+            Query = "query helloQuery { hello }",
+            RequestServices = _host.Services,
+        };
+        var ranFilter = false;
+        _options.Filter = options =>
+        {
+            options.ShouldBe(executionOptions);
+            ranFilter = true;
+            return false;
+        };
+
+        // execute GraphQL document
+        var result = await _executer.ExecuteAsync(executionOptions).ConfigureAwait(false);
+
+        // verify GraphQL response
+        _serializer.Serialize(result).ShouldBe("""{"data":{"hello":"World"}}""");
+
+        // verify activity telemetry
+        _exportedActivities.ShouldBeEmpty();
+        ranFilter.ShouldBeTrue();
+    }
+
+    [Fact]
+    public async Task WithValidationError()
```

Comment:
Maybe add tag like "graphql.validation_errors" in case of failed validations?

Reply:
I would not at this time.  Maybe opt-in through an options property.  The concept is collecting aggregate telemetry not diagnostics.

Reply:
OK